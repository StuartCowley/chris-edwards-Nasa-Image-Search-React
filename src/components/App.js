import React, { useState } from "react";
import SearchBar from "./Search";
import "../styles/app.css";
import SearchResults from "./SearchResults";
import Tags from "./Tags"; // Import the Tags component

function App() {
  const [results, setResults] = useState([]);

  return (
    <div className="App">
      <img
        className="logo"
        src="https://99designs-blog.imgix.net/blog/wp-content/uploads/2019/03/2000px-NASA_Worm_logo.svg.png?auto=format&q=60&fit=max&w=930"
        alt="Nasa Logo"
      />
      <SearchBar setSearchResults={setResults} />
      <Tags setSearchResults={setResults} />
      <SearchResults results={results} />
    </div>
  );
}

export default App;
