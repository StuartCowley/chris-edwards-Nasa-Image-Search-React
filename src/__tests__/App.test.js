import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders NASA logo", () => {
    render(<App />);
    const logoElement = screen.getByAltText(/Nasa Logo/i);
    expect(logoElement).toBeInTheDocument();
  });
});
