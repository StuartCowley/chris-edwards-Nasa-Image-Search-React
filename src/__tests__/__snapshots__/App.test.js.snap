// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="Nasa Logo"
      class="logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <div
      class="search-bar-container"
    >
      <form>
        <input
          placeholder="Search..."
          type="text"
          value=""
        />
        <button
          class="search-button"
          type="submit"
        >
          Blast off
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
